components/supervisor/pkg/terminal/terminal.go

```diff
@@ -176,11 +196,24 @@ func (l *multiWriterListener) Done() <-chan struct{} {
 	return l.closeChan
 }
 
+type closedTerminalListener struct {
+}
+
+func (closedTerminalListener) Read(p []byte) (n int, err error) {
+	return 0, errors.New("terminal is closed")
```

Comment:
we're implementing `io.Reader` whose contract mandates we return `io.EOF` in this case.

```
return 0, io.EOF
```